### ISSUE TITLE
Upgrade batches naming

### ISSUE THREAD

**Issue by MelLain** [bigartm/bigartm#397]
Currently BigARTM parser creates bathes with uuid names. It's very unwanted behavior: the model, built on one set of batches, can't be repeated on another set, because each set has it's own documents order, depends on the alphabet order of batches names.

The solution is to create batches with names NNNNNN.batch, where N is in {'A', ..., 'Z'} in the natural order. This gives an opportunity to create approximately 300 million batches, and the order of the documents in them always will be the same (1, 2, etc.)

**Comment by sashafrey** [comment]
I agree this is important to generate deterministic names from run to run. However, I just realize that this will break one important scenario where batches are generated in parallel (in several `bigartm.exe` processes), and the results are stored in a single folder. Currently it is very convenient that BigARTM will always generate unique names for each batch. We need to figure out a solution that works for both cases.

**Comment by sashafrey** [comment]
My suggestion is to generate a deterministic unique name based on the content of the batch:
```
batch.set_id("00000000-0000-0000-0000-000000000000")
std::string str = batch.serializeToString();
std::string md5 = ...  // find MD5 hash of str)
boost::uuids::uuid dns_namespace_uuid; // an arbitrary GUID, hardcoded once and forever
boost::uuids::name_generator gen(dns_namespace_uuid);
boost::uuids::uuid u = gen(str.c_str()); 
batch.set_id(u);
saveToFile(batch, u);
```

This uses feature called *name based GUID* --- see here: http://stackoverflow.com/questions/2642141/how-to-create-deterministic-guids. This is implemented in boost, http://www.boost.org/doc/libs/1_57_0/libs/uuid/uuid.html#Name Generator. 

A good question is how to has a content of a batch. I think the easiest way is to serialize it to string, and then find MD5 (or another standard way, whatever boost can offer).

**Comment by sashafrey** [comment]
@MelLain Actually my suggestion is also bad, because it doesn't preserve the order. Maybe some users want to iterate through batches in the same order as they appear in vw- or docword- files.

**Comment by sashafrey** [comment]
We agreed on keeping both options:
```
message CollectionParserConfig {
  ...
  enum BatchName {
    Guid,
    Latin,  // or Code
  }

  optional BatchName batch_name [default = Guid];
}
```

**Comment by sashafrey** [comment]
This will be addressed by PR #400

**Comment by sashafrey** [comment]
Fixed.
